cli/internal/cache/cache_fs.go

```diff
@@ -85,40 +88,19 @@ func (f *fsCache) logFetch(hit bool, hash string, duration int) {
 }
 
 func (f *fsCache) Put(anchor turbopath.AbsoluteSystemPath, hash string, duration int, files []turbopath.AnchoredSystemPath) error {
-	g := new(errgroup.Group)
-
-	numDigesters := runtime.NumCPU()
-	fileQueue := make(chan turbopath.AnchoredSystemPath, numDigesters)
-
-	for i := 0; i < numDigesters; i++ {
-		g.Go(func() error {
-			for file := range fileQueue {
-				statedFile := fs.LstatCachedFile{Path: file.RestoreAnchor(anchor)}
-				fromType, err := statedFile.GetType()
-				if err != nil {
-					return fmt.Errorf("error stat'ing cache source %v: %v", file, err)
-				}
-				if !fromType.IsDir() {
-					if err := f.cacheDirectory.UntypedJoin(hash, file.ToStringDuringMigration()).EnsureDir(); err != nil {
-						return fmt.Errorf("error ensuring directory file from cache: %w", err)
-					}
-
-					if err := fs.CopyFile(&statedFile, f.cacheDirectory.UntypedJoin(hash, file.ToStringDuringMigration()).ToStringDuringMigration()); err != nil {
-						return fmt.Errorf("error copying file from cache: %w", err)
-					}
-				}
-			}
-			return nil
-		})
-	}
+	cachePath := f.cacheDirectory.UntypedJoin(hash + ".tar.gz")
+	cacheItem, err := cacheitem.Create(cachePath)
+	defer func() { _ = cacheItem.Close() }()
```

Comment:
This seems like the kind of error we might care about. If we fail to close, should we consider it as not-actually-stored?

Regardless, it should go after the error check for `Create`. If we failed to `Create`, we don't need to `Close`

Reply:
You're right, we should actually inspect the close error. Failing `Close` doesn't _precisely_ mean it can't be restored, but we shouldn't attempt it. (You lose the last 1024 bytes of 0x0 padding and the trailing gzip checksum.)

Ordering: my habit for deferrals is "immediately after" so that I don't accidentally do it wrong. Keeping them together prevents mistakes. (There's a later commit in this PR where I failed to do that in tests and only closed later which was accidentally conditional and left a bug. 🤦‍♂️)